src/components/datepicker/js/calendar.js

```diff
@@ -205,18 +200,41 @@
       angular.element(document.body).off('keydown', boundKeyHandler);
     });
 
-    if (this.minDate && this.minDate > $mdDateLocale.firstRenderableDate) {
+    // For older Angular versions, where the bindings are pre-assigned, manually call the $onInit hook.
```

Comment:
This sounds like you're calling `$onInit` manually because bindings are pre-assigned and the real reason is that this hook is not supported in older versions and that's why you have to call it. The fact that bindings are pre-assigned only means that it will work even though normally the hook would be invoked later. How about:
> For Angular 1.4 and older, where there are no lifecycle hooks but bindings are pre-assigned, manually call the $onInit hook.